homeassistant/components/verisure/lock.py

```diff
@@ -98,6 +98,11 @@ def available(self) -> bool:
     def changed_by(self) -> str | None:
         """Last change triggered by."""
         return self.coordinator.data["locks"][self.serial_number].get("userString")
+    
+    @property
+    def changed_method(self) -> str | None:
+        """Last change method."""
+        return self.coordinator.data["locks"][self.serial_number].get("method")
```

Comment:
Good point. I don't have access any other installations than mine, and the API seems undocumented. In my tests in my system "method" have always been present, but of course I cannot know for sure for other installations. If it is not present changed_method will return None.

If changed_method is None, extra_state_attributes will return {'method': None} I assume HA will handle attributes with NoneType? Or would it be better if extra_state_attributes returned an empty dict if "method" is not present?

Reply:
The entity can handle a `None` value for a state attribute. It would be good to know the possible return values here so we can type it as narrow as possible.

Reply:
I would make it narrow and widen if needed (this part is simply not documented, so no way to know).

Reply:
I'm not sure if I fully understand what you mean by narrow? 'method' will be a string if present or None if not present. I listed the possible values of "method" I have found in my installation in the description of the pull request:

"method": "THUMB",
"method": "STAR",
"method": "CODE",
"method": "REMOTE",
"method": "AUTO",

Reply:
If we think it will always be present, we can change it like so:
```suggestion
        return self.coordinator.data["locks"][self.serial_number]["method"]
```
And update the return value type to just `str`.

Reply:
To me that seems a bit risky since the Verisure interface is not documented. If the "method" is not present there will be an exception with your proposed edit. To me it seems better to get `None` if it is not present in some installations. I only have access to mine which is only one of the 722 active Verisure HA installations, so I have no way of knowing that "method" will always be present in all intergrations. But like I'm said I'm new to HA so I respect the decision of the more experienced than me.

Reply:
> To me that seems a bit risky since the Verisure interface is not documented.

We'll be able to patch it out quickly enough if this is an issue. It is easier to loosing later than to make it more strict later...